meshpy/FourC/beam_potential.py

```diff
@@ -0,0 +1,256 @@
+            REGULARIZATION_SEPARATION       {regularization_separation}"""
+            )
+        if choice_master_slave is not None:
+            settings.append(f"\nCHOICE_MASTER_SLAVE             {choice_master_slave}")
+
+        self.input_file.add(
+            InputSection(
+                "BEAM POTENTIAL",
+                settings,
+                option_overwrite=option_overwrite,
+            )
+        )
+
+    def add_runtime_output(
+        self,
+        *,
+        output_beam_potential=True,
+        interval_steps=1,
+        every_iteration=False,
+        forces=True,
+        moments=True,
+        per_ele_pair=True,
+        option_overwrite=False,
+    ):
+        """
+        Set the basic runtime output options for beam potential interactions.
+
+        Args
+        ----
+        output_beam_potential: bool
+            If the output for beam potential should be written.
+        interval_steps: int
+            Interval at which output is written.
+        every_iteration: bool
+            If output at every Newton iteration should be written.
+        forces: bool
+            If the forces should be written.
+        moments: bool
+            If the moments should be written.
+        per_ele_pair: bool
+            If the forces/moments should be written per element pair.
+        option_overwrite: bool
+            If existing options should be overwritten. If this is false and an
+            option is already defined, and error will be thrown.
+        """
+
+        self.input_file.add(
+            InputSection(
+                "BEAM POTENTIAL/RUNTIME VTK OUTPUT",
+                f"""
+            VTK_OUTPUT_BEAM_POTENTIAL           {get_yes_no(output_beam_potential)}
+            INTERVAL_STEPS                      {get_yes_no(interval_steps)}
+            EVERY_ITERATION                     {get_yes_no(every_iteration)}
+            FORCES                              {get_yes_no(forces)}
+            MOMENTS                             {get_yes_no(moments)}
+            WRITE_FORCE_MOMENT_PER_ELEMENTPAIR  {get_yes_no(per_ele_pair)}""",
+                option_overwrite=option_overwrite,
+            )
+        )
+
+    def add_potential_charge_condition(self, *, geometry_set=None):
+        """Add potential charge condition to geometry.
+
+        Args
+        ----
+        geometry_set:
+            Add potential charge condition to this set.
+        """
+
+        for i, (line_charge, func) in enumerate(
+            zip(
+                self.pot_law_line_charge_density, self.pot_law_line_charge_density_funcs
+            )
+        ):
+
+            if func != "none":
+                self.input_file.add(func)
+
+            bc = BoundaryCondition(
+                geometry_set,
+                f"POTLAW {i+1} VAL {line_charge} FUNCT {str(func)}",
+                bc_type="DESIGN LINE BEAM POTENTIAL CHARGE CONDITIONS",
+            )
```

Comment:
```suggestion
            bc = BoundaryCondition(
                geometry_set,
                f"POTLAW {i+1} VAL {line_charge} FUNCT {{}}",
                bc_type="DESIGN LINE BEAM POTENTIAL CHARGE CONDITIONS",
                format_replacement=[func],
            )
```
The `BoundaryCondition` object has the functionality to evaluate certain string replacements only when the actual line for the input file is created. In other words, the `InputFile` can create the ordering and indexing of the functions and then the boundary condition automatically inserts the correct function ID. This mechanism is not perfect, but works quire well for the replacement of function IDs in BC definitions. This should fix the issue you raised in https://github.com/imcs-compsim/meshpy/pull/74#issuecomment-2093084025. I think in the case of `func="none"` this should also work, but please make a quick test to verify this.

Reply:
Thanks for this suggestion! It works like a charm, both for defined functions, as well as the `none` case.